### PR TITLE
Feature/dep updates

### PR DESCRIPTION
**Pull request by st3v3nhunt**
Supersedes #232

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
     "coverage-clear": "rm -Rf coverage",
     "coverage-check": "yarn coverage-generate && istanbul check-coverage --config .istanbul.yml",
     "coverage-check-unit": "yarn coverage-generate-unit && istanbul check-coverage --config .istanbul.unit.yml",
-    "coverage-generate": "yarn coverage-clear && yarn istanbul cover -- _mocha -- --recursive",
-    "coverage-generate-unit": "yarn coverage-clear && yarn istanbul cover -- _mocha -- --recursive test/unit",
+    "coverage-generate": "yarn coverage-clear && yarn istanbul -- cover _mocha -- --recursive",
```
Comment (beseven) [inline package.json]: Not sure why this change is required as based on this https://github.com/gotwarlost/istanbul#the-cover-command the `old` way looks fine.

**Comment by st3v3nhunt** [comment]
I've specifically left the last 2 commits to capture the reasoning for not upgrading to node 8.6.0

File: package.json (hunk 1 above)
Comment (st3v3nhunt) [reply package.json]: My bad, this needs sorting.

**Comment by beseven** [comment]
Approved bar the `upgrading` snyk part. Let me know what you want to do about that, @st3v3nhunt.
